File: coderbyte/a_easy/javascript/07_simple_symbols/simple_symbols.js
const simpleSymbols = (str) => {
  const bools = [];
  const array = str.split('');

  if (array.length <= 2) return false;

  for (let i = 0; i < str.length; i++) {
    if (/[A-Za-z]/.test(array[i]) && i !== 0 && i !== array.length) {
      if (array[i - 1] === '+' && array[i + 1] === '+') {
        bools.push(true);
      } else {
        bools.push(false);
      }
    }
  }

  return !bools.includes(false);
};

module.exports = simpleSymbols;

File: coderbyte/a_easy/ruby/07_simple_symbols/simple_symbols.rb
def simple_symbols(str)
  bools = []

  for n in 0..str.length - 1 do
    if str[n].match(/[A-Za-z]/)
      if n.zero?
        bools << false
      elsif n == str.length - 1
        bools << (str[n - 1] == '+')
      else
        bools << (str[n - 1] == '+' && str[n + 1] == '+')
      end
    end
  end

  bools.all?
end
